plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/SingletonIdentityCacheMapping.java

```diff
@@ -24,21 +24,9 @@ public IdentityCacheKey getRemoteUserCacheKey(ConnectorSession session)
         return SingletonIdentityCacheKey.INSTANCE;
     }
 
-    private static final class SingletonIdentityCacheKey
+    private enum SingletonIdentityCacheKey
             implements IdentityCacheKey
     {
-        private static final SingletonIdentityCacheKey INSTANCE = new SingletonIdentityCacheKey();
-
-        @Override
-        public int hashCode()
-        {
-            return 0;
-        }
-
-        @Override
-        public boolean equals(Object obj)
-        {
-            return obj instanceof SingletonIdentityCacheMapping;
```

Comment:
How about doing `obj == INSTANCE`? Without going with enum?

Reply:
OK, I went back to it being an abstract class, and comparing it by identity (to `this`, though, because otherwise IDEA complain that we should check the type as well).

Reply:
So there are two versions, I can go with any one of them.

Reply:
I would keep the abstract class.